pswalker/iterwalk.py

```diff
@@ -221,27 +230,39 @@ def iterwalk(detectors, motors, goals, starts=None, first_steps=1,
                 else:
                     goal = (goals[index] - pos) * (1 + overshoot) + pos
 
+                # Calculate adaptive tolerence - otherwise use static tolerence
+                if tol_scaling[index] != None:
+                    selected_tol[index] = \
+                        abs(pos-goals[index]) / tol_scaling[index]
+                    if selected_tol[index] < tolerances[index]:
+                        selected_tol = tolerances[index]
```

Comment:
This line could break the code by changing the type of selected_tol from list to int. Shouldn't it be:
`selected_tol[index] = tolerances[index]`